network/p2p/blob_service.go

```diff
@@ -84,9 +86,34 @@ func NewBlobService(
 
 	cm := component.NewComponentManagerBuilder().
 		AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
-			bs.blockService = blockservice.New(bs.blockStore, bitswap.New(ctx, bsNetwork, bs.blockStore, bs.config.BitswapOptions...))
+			btswp := bitswap.New(ctx, bsNetwork, bs.blockStore, bs.config.BitswapOptions...).(*bitswap.Bitswap)
+			bs.blockService = blockservice.New(bs.blockStore, btswp)
 
 			ready()
+
+			ticker := time.NewTicker(15 * time.Second)
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case <-ticker.C:
+					stat, err := btswp.Stat()
+					if err != nil {
+						// TODO: log error
+						continue
+					}
+
+					metrics.Peers(prefix, len(stat.Peers))
```

Comment:
it looks like prefix should be a property of the BitswapCollector instead of being passed to every function call.